packages/zendesk-client/src/use-smooch.ts

```diff
@@ -0,0 +1,121 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { useState } from 'react';
+import Smooch from 'smooch';
+import { SMOOCH_INTEGRATION_ID } from './constants';
+import { UserFields } from './types';
+import { useAuthenticateZendeskMessaging } from './use-authenticate-zendesk-messaging';
+import { useUpdateZendeskUserFields } from './use-update-zendesk-user-fields';
+
+export const useSmooch = () => {
+	const [ init, setInit ] = useState( typeof Smooch.getConversations === 'function' );
+	const { data: authData } = useAuthenticateZendeskMessaging( true, 'messenger' );
+	const { isPending: isSubmittingZendeskUserFields, mutateAsync: submitUserFields } =
+		useUpdateZendeskUserFields();
+
+	const initSmooch = ( ref: HTMLDivElement ) => {
+		if ( authData?.jwt && authData?.externalId && ! Smooch.getConversations ) {
+			Smooch.init( {
+				integrationId: SMOOCH_INTEGRATION_ID,
+				embedded: true,
+				externalId: authData?.externalId,
+				jwt: authData?.jwt,
+			} )
+				.then( () => {
+					recordTracksEvent( 'calypso_smooch_messenger_init', {
+						success: true,
+						error: '',
+					} );
+					setInit( true );
+				} )
+				.catch( ( error ) => {
+					recordTracksEvent( 'calypso_smooch_messenger_init', {
+						success: false,
+						error: error.message,
+					} );
+					setInit( false );
+				} );
+			Smooch.render( ref );
+		}
+	};
+
+	const destroy = () => {
+		Smooch.destroy();
+	};
+
+	if ( ! init ) {
+		return {
+			init,
+			destroy,
+			initSmooch,
+			getConversation: async () => undefined,
+			createConversation: async () => undefined,
+			addMessengerListener: () => undefined,
+			addUnreadCountListener: () => undefined,
+			sendMessage: () => undefined,
+		};
+	}
+
+	const getConversation = async ( chatId?: number ): Promise< Conversation | undefined > => {
+		if ( chatId ) {
+			const existingConversation = Smooch.getConversations?.().find( ( conversation ) => {
+				return conversation.metadata[ 'odieChatId' ] === chatId;
+			} );
+			if ( ! existingConversation ) {
+				return;
+			}
+			const result = await Smooch.getConversationById( existingConversation.id );
+			if ( result ) {
+				Smooch.markAllAsRead( result.id );
+				return result;
+			}
+		}
+		return;
+	};
+
+	const createConversation = async (
+		userfields: UserFields,
+		metadata: Conversation[ 'metadata' ]
+	) => {
+		if ( isSubmittingZendeskUserFields ) {
+			return;
+		}
+		if ( metadata.odieChatId ) {
+			await submitUserFields( userfields );
+			await Smooch.createConversation( { metadata } );
```

Comment:
We need it to be sequential, since `submitUserFields` will fill the details required to link the new Zendesk ticket with the Odie chat (and a couple more props)